modules/expo-bluesky-translate/android/src/main/java/expo/modules/blueskytranslate/ExpoBlueskyTranslateModule.kt

```diff
@@ -0,0 +1,65 @@
+package expo.modules.blueskytranslate
+
+import com.google.mlkit.common.model.DownloadConditions
+import com.google.mlkit.nl.translate.Translation
+import com.google.mlkit.nl.translate.TranslatorOptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import expo.modules.kotlin.Promise
+
+class ExpoBlueskyTranslateModule : Module() {
+    var translationPromise: Promise? = null
+
+    override fun definition() = ModuleDefinition {
+        Name("ExpoBlueskyTranslate")
+
+        AsyncFunction("translateAsync") { sourceLanguage: String, targetLanguage: String, text: String, promise: Promise ->
+            if (translationPromise != null) {
+                promise.reject(TranslationInProgressException())
+                return@AsyncFunction
+            }
+
+            translationPromise = promise
+
+            val options = TranslatorOptions.Builder()
+                .setSourceLanguage(sourceLanguage)
+                .setTargetLanguage(targetLanguage)
+                .build()
+
+            val translator = Translation.getClient(options)
+
+            val conditions = DownloadConditions.Builder()
+                .requireWifi()
+                .build()
+
+
+            translator.downloadModelIfNeeded(conditions)
+                .addOnSuccessListener {
+                    translator.translate(encodeSpecialCharacters(text))
+                        .addOnSuccessListener { translatedText ->
+                            promise.resolve(decodeSpecialCharacters(translatedText))
+                            translationPromise = null
+                        }
+                        .addOnFailureListener { exception ->
+                            promise.reject(UnableToTranslateTextException())
+                            translationPromise = null
+                        }
+                }
+                .addOnFailureListener { exception ->
+                    promise.reject(UnableToTranslateTextException())
+                    translationPromise = null
+                }
+
+        }
+    }
+
+  private fun encodeSpecialCharacters(text: String): String {
+    return text.replace("\n", "␤")  // Use Unicode character for newline
```

Comment:
in cases where the posts contain formatted text, this is a best-effort attempt to preserve this formatting after translation (the translator will strip whitespace characters otherwise)